Ignore public suffixes when checking whether to apply adjustments in `Document::urlForBindings`

whsieh opened the pull request:
#### 777ad4c846a81579abb195e0e87b52b6ccd4b6a1
<pre>
Ignore public suffixes when checking whether to apply adjustments in `Document::urlForBindings`
<a href="https://bugs.webkit.org/show_bug.cgi?id=256329">https://bugs.webkit.org/show_bug.cgi?id=256329</a>
rdar://108691950

Reviewed by Aditya Keerthi.

In a similar vein to 262965@main, discount public suffixes when determining whether or not we should
adjust `urlForBindings`, as long as the top level domain otherwise matches. This patch pulls that
previously introduced logic out into a separate lambda function, and consults it in both places.

* Source/WebCore/dom/Document.cpp:
(WebCore::Document::urlForBindings const):

Canonical link: <a href="https://commits.webkit.org/263711@main">https://commits.webkit.org/263711@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/7ef2f898f5919d16a68bf3c68d0be1de44bf18c4

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/5505 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/5651 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/5841 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/7050 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/5514 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/5503 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/5876 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/5625 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/6803 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/5606 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/5665 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/4926 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/7084 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/3142 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/4941 "Passed tests") | [⏳ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/GTK-WK2-Tests-EWS "Waiting to run tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/5008 "Passed tests") | [❌ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/5021 "3 failures") | [⏳ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/API-Tests-GTK-EWS "Waiting to run tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/5456 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/17/builds/4470 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/4912 "Built successfully") | | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/1315 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/9015 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/5274 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

aprotyas commented on Source/WebCore/dom/Document.cpp:
```diff
@@ -3585,29 +3585,35 @@ const URL& Document::urlForBindings() const
+                if (indexOfFirstSeparator == notFound)
+                    return { };
+                return string.left(indexOfFirstSeparator);
+            };
```
Did you have to capture anything in this lambda?

whsieh replied:
Nope, good catch! Removed the `&`.

sysrqb reviewed (approved):
lgtm.

whsieh committed:
Ignore public suffixes when checking whether to apply adjustments in `Document::urlForBindings`
https://bugs.webkit.org/show_bug.cgi?id=256329
rdar://108691950

Reviewed by Aditya Keerthi.

In a similar vein to 262965@main, discount public suffixes when determining whether or not we should
adjust `urlForBindings`, as long as the top level domain otherwise matches. This patch pulls that
previously introduced logic out into a separate lambda function, and consults it in both places.

* Source/WebCore/dom/Document.cpp:
(WebCore::Document::urlForBindings const):

Canonical link: https://commits.webkit.org/263711@main
